### PR TITLE
Added sample method to DashAIDataset

### PR DESCRIPTION
**Pull request by nataliabarca**
# Summary
Added a sample method to DashAIDataset class. This method generates samples from a dataset with three different options: retrieve the first n rows, the last n rows or n random rows.

The purpose of this change is to be able to retrieve samples from a dataset when using the Prediction tab to show them to the user.

## Changes
- Added sample method to DashAIDataset in `dashai_dataset.py`
- Added new test in `test_dashai_dataset.py`

## How to Test
Run `pytest ./tests/back/dataloaders/test_dashai_dataset.py`

### CODE REVIEW

File: DashAI/back/dataloaders/classes/dashai_dataset.py [hunk 1]
```diff
@@ -120,6 +121,49 @@ def change_columns_type(self, column_types: Dict[str, str]):
         dataset = self.cast(new_features)
         return dataset
 
+    def sample(
+        self,
+        n: int = 1,
+        method: Literal["head", "tail", "random"] = "head",
+        seed: int | None = None,
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/classes/dashai_dataset.py]: This notation is not compatible with python 3.8 :((

You have to change it to:

```suggestion
        seed: Union[int, None] = None,
```

And import as `from typing import Union`

File: DashAI/back/dataloaders/classes/dashai_dataset.py [hunk 2]
```diff
@@ -120,6 +121,49 @@ def change_columns_type(self, column_types: Dict[str, str]):
...
+    ):
```
Comment (pbadillatorrealba) [inline DashAI/back/dataloaders/classes/dashai_dataset.py]: Question: what kind of object returns, a python dict, dashai dataset or huggingface dataset?

Please add a return type to make this explicit

File: tests/back/dataloaders/test_dashai_dataset.py [hunk 3]
```diff
@@ -94,6 +94,27 @@ def fixture_dashaidataset():
     return [datasetdict, dataloader_test]
 
 
+def test_dashaidataset_sample(dashaidataset_created: list):
```
Comment (pbadillatorrealba) [inline tests/back/dataloaders/test_dashai_dataset.py:97]: Could you also test for n!=1?

**Review by pbadillatorrealba — CHANGES_REQUESTED** [review]
Some minor comments about types and tests

**Comment by nataliabarca** [comment]
All requested changes are completed 👌

**Comment by pbadillatorrealba** [comment]
Nice!

I guess that merging this pr would not affect the current release, what do you think @cristian-tamblay ?

**Comment by cristian-tamblay** [comment]
Good to merge!

**Pull request merged**